### DISCUSSION TITLE
Data is not converted into array when with the DB facade.

### DISCUSSION THREAD

**Discussion by tamdaz** [laravel/framework#53658] [Q&A] [answered]
### Laravel Version

11.33.2

### PHP Version

8.2.25

### Database Driver & Version

MariaDB ver. 10.11.10

### Description

Hello,
I have encountered an inconsistency when converting data into a array using the Eloquent model and the DB facade. When I use the Eloquent model, it converts the data into an array.
However, with the DB facade, all the data is represented as objects and it's not possible to convert them into arrays using the `toArray()` method.

# With Model
```php
// Using model
dump(Project::all()->toArray());
```

```
// The `Project` model converts data into arrays.
array:2 [
  0 => array:8 [
    "id" => 1
    "title" => "Acme"
    "slug" => "acme"
    "description" => "Acme is an example project"
    "content" => "Lorem ipsum dolor sit amet consectur."
    "category" => "Personal projects"
    "realized_at" => "2024-12-31T23:00:00.000000Z"
    "published_at" => null
  ]
  1 => array:8 [
    "id" => 2
    "title" => "Studentus"
    "slug" => "studentus"
    "description" => "A very short description"
    "content" => "A very long content"
    "category" => "Personal projects"
    "realized_at" => "2025-02-28T23:00:00.000000Z"
    "published_at" => null
  ]
] // app/Infrastructure/Console/Commands/ManageProjectVersion.php:46
```

```php
// Using DB facade
dump(DB::table('projects')->get()->toArray());
```

# With DB Facade

```
// But this data is not converted into arrays with DB facade.
array:2 [
  0 => {#1861
    +"id": 1
    +"title": "Acme"
    +"slug": "acme"
    +"description": "Acme is an example project"
    +"content": "Lorem ipsum dolor sit amet consectur."
    +"category": "Personal projects"
    +"realized_at": "2025-01-01"
    +"published_at": null
  }
  1 => {#1109
    +"id": 2
    +"title": "Studentus"
    +"slug": "studentus"
    +"description": "A very short description"
    +"content": "A very long content"
    +"category": "Personal projects"
    +"realized_at": "2025-03-01"
    +"published_at": null
  }
] // app/Infrastructure/Console/Commands/ManageProjectVersion.php:50
```

### Steps To Reproduce

To make comparisons, I had to display this data in the class that extends Console.

```php  
/**
 * Execute the console command.
 */
public function handle(): int
{
	// Using model
	dump(Project::all()->toArray());

	// Using DB facade
	dump(DB::table('projects')->get()->toArray());

	return Command::SUCCESS;
}
```

This doesn't cause bugs, but rather inconsistencies.

**Comment by rodrigopedra** [comment, marked as answer]
That is by design.

When you call `Collection@toArray()`, the collection instance iterates over its elements and checks if they implement the `Illuminate\Support\Traits\Arrayable` interface. If so, it calls that element's `toArray()` method.

When you query your DBMS through Eloquent, the result set is hydrated into Eloquent models. Those instances implement the `Illuminate\Support\Traits\Arrayable`, and thus know how to convert themselves into arrays when needed.

When you query your DBMS through the DB façade, the result set is not changed, and by default each record will be an instance of PHP's built-in `\stdClass`, and as such, those do not implement any of Laravel's interfaces.

So, when the collection iterates over these records, it cannot ask them for their array representation.

If you need, for some reason, to convert those records to an array, you can try this:

```php
dump(DB::table('projects')->get()->map(fn ($record) => (array) $record)->toArray());
```

The `map()` operation will cast those records natively to arrays, before the `toArray()` call.

**Comment by rishabhtpt** [comment]
[Data models in DBMS](https://www.tpointtech.com/data-models) (Database Management Systems) define how data is structured, stored, and related. The main types are hierarchical, network, relational, and object-oriented models. Among these, the relational model is most widely used due to its simplicity and flexibility. Data models help organize information clearly and make it easier to manage, query, and update databases efficiently.
